### PR TITLE
Print CPU flags at startup

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Example: 
`2024.02.16 13:04:31.423643 [ 2394820 ] {} <Information> Application: Available CPU instruction sets: SSE, SSE2, SSE3, SSSE3, SSE41, SSE42, F16C, POPCNT, BMI1, BMI2, PCLMUL, AES, AVX, FMA, AVX2, SHA, ADX, RDRAND, RDSEED, RDTSCP, CLFLUSHOPT, CLWB, XSAVE, OSXSAVE`

See https://github.com/ClickHouse/ClickHouse/issues/60017 which is probably related to the lack of __msan_unpoison after using AVX-512.

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 39f363ba6e7dc21c1ac9d8e1dddc92d677e5b302 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60075/39f363ba6e7dc21c1ac9d8e1dddc92d677e5b302/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60075/39f363ba6e7dc21c1ac9d8e1dddc92d677e5b302/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by devcrafter — APPROVED** [review]
LGTM, but build needs to be fixed, - CPU profile events are used in Keeper as well (`src/Coordination/KeeperConstants.cpp`)

@antonio2368 probably some clarifications, - why do we need to declare profile events in `KeeperConstants.cpp`? It doesn't look like necessary

**Comment by devcrafter** [comment]
>  why do we need to declare profile events in KeeperConstants.cpp? It doesn't look like necessary

It looks it was done for standalone keeper

**Comment by antonio2368** [comment]
> why do we need to declare profile events in KeeperConstants.cpp? It doesn't look like necessary

@devcrafter the idea is to avoid sending server related metrics from Keeper's prometheus so we manually define metrics we want to have.
I didn't find any easier way for now, I could maybe do something with macros in ProfileEvents.cpp but I'm afraid of overcomplicating code around it.

File: programs/server/Server.cpp [hunk 1]
```diff
@@ -712,6 +713,20 @@ try
...
+    {                   \
+        if (!cpu_info.empty()) \
+            cpu_info += ", ";  \
+        cpu_info += #X; \
+    }
+
+    CPU_ID_ENUMERATE(COLLECT_FLAG)
+#undef COLLECT_FLAG
+
+    LOG_INFO(log, "Available CPU instruction sets: {}", cpu_info);
```
Comment (devcrafter) [inline programs/server/Server.cpp:728]: It looks like we would print something only on x86 family processors. Then the message for non x86 will not be useful

**Pull request merged**